pulsar/dlq_router.go

```diff
@@ -135,12 +135,16 @@ func (r *dlqRouter) getProducer(schema Schema) Producer {
 	// Retry to create producer indefinitely
 	backoff := &internal.Backoff{}
 	for {
-		producer, err := r.client.CreateProducer(ProducerOptions{
-			Topic:                   r.policy.DeadLetterTopic,
-			CompressionType:         LZ4,
-			BatchingMaxPublishDelay: 100 * time.Millisecond,
```

Comment:
Is there a reason we disabled batching by default?

Reply:
Batching is enabled by default. The `DisableBatching` in the ProducerOptions is false by the default. So the producer for DLQ and RLQ is always batched.

Reply:
So just to be explicit, this effectively changes the `BatchingMaxPublishDelay` from 100 millis to the producer default, which is first 0 millis and then is overridden to 10 millis in the producer's constructor. Given that this would otherwise require merging producer options, I think this is fine as long as we document it in the release notes.